### PR TITLE
Show stats v4 by default if possible

### PR DESCRIPTION
**Pull request by Ecarrion**
# Why
As part of https://github.com/woocommerce/woocommerce-ios/issues/1946 we will be updating the stats banner to let the user know that the v3 stats will be deprecated soon.

This PR changes the behavior to the following:
- If stats v4 are available, we default to show them and show no banner
- If stats v4 are not available, we show stats v3 dashboards and show a collapsable banner indicating the user that the v3 stats will be deprecated soon.

Changes in following PRs will include:
- Dismiss and remind me later actions
- Empty state screen if the user hasn't updated it's Woo version and September 1st has arrived.

# How

First, we have reduced the `StatsVersionState` enum to only have one case(`initial`) because we don't care anymore about transitioning between v4 or v3 states. In the future we aim to remove this enum and only store it's associated property `statsVersion`. While doing that we have also trimmed a lot of tests that weren't needed anymore. 

Second, we have added a new banner using `TopBannerView` to indicate the user that they are using and old and soon to be deprecated version of the stats. With this we have deleted the code associated with `V3ToV4` and `V4ToV3` banners.

# Screenshots

V3 | V4
--- | ---
v3|v4

# Testing steps
- Run the app using a store that doesn't have v4 stats available and see that the correct banner(indicating the user to upgrade) is being shown.
- Run the app using a store that has v4 stats available and see that the new stats dashboard is being shown and not banner is present.

# Questions
- The android designs have a black tone for the icon on the banner, other banners on the iOS app have a gray-ish tone, should we keep the gray tone for this banner too @Garance91540 ?

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/WooCommerceTests/Testing/XCTestCase+Wait.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    /// waitForExpectation(timeout: TimeInterval(10)) { expectation in
+    ///     doSomethingInTheBackground {
+    ///         expectation.fulfill()
+    ///     }
+    /// }
+    /// ```
+    ///
+    func waitForExpectation(description: String? = nil,
+                            timeout: TimeInterval = Constants.expectationTimeout,
+                            _ block: (XCTestExpectation) -> ()) {
```
Comment (jaclync) [inline WooCommerce/WooCommerceTests/Testing/XCTestCase+Wait.swift:19]: thanks for adding this!

**Comment by shiki** [comment]
> Just double checking, is the removal of v4 stats feature switch under Settings > Experimental Features a separate task?

@jaclync That's right. We're going to work on it in #2452.

**Comment by Ecarrion** [comment]
> When there's no experimental feature available, instead of opening a blank screen, can we make the "Experimental Features" link disabled on the Settings page?

@Garance91540 How do you feel about hiding the "Experimental Features" row rather tan disabling it? I think that is more inline with how apple manages the settings on the phone.

**Comment by Garance91540** [comment]
> @Garance91540 How do you feel about hiding the "Experimental Features" row rather tan disabling it? I think that is more inline with how apple manages the settings on the phone.

That works!

**Comment by Ecarrion** [comment]
@Garance91540 I've updated the view to match the figma design. The red dot will be implemented in a different PR https://github.com/woocommerce/woocommerce-ios/issues/2465
new-aligment-text

Please review again :-)

**Pull request merged**